web/web.go

```diff
@@ -529,23 +531,52 @@ func signupCompleteOAuth(c *api.Context, w http.ResponseWriter, r *http.Request)
 			return
 		}
 
-		if result := <-api.Srv.Store.User().GetByAuth(team.Id, user.AuthData, service); result.Err == nil {
+		suchan := api.Srv.Store.User().GetByAuth(team.Id, user.AuthData, service)
+		euchan := api.Srv.Store.User().GetByEmail(team.Id, user.Email)
+
+		if team.Email == "" {
+			team.Email = user.Email
+			if result := <-api.Srv.Store.Team().Update(team); result.Err != nil {
+				c.Err = result.Err
+				return
+			}
+		} else {
+			found := true
+			count := 0
+			for found {
+				if found = api.IsUsernameTaken(user.Username, team.Id); c.Err != nil {
+					return
+				} else if found {
+					user.Username = user.Username + strconv.Itoa(count)
+					count += 1
+				}
+			}
+		}
+
+		if result := <-suchan; result.Err == nil {
 			c.Err = model.NewAppError("signupCompleteOAuth", "This "+service+" account has already been used to sign up for team "+team.DisplayName, "email="+user.Email)
 			return
 		}
 
-		if result := <-api.Srv.Store.User().GetByEmail(team.Id, user.Email); result.Err == nil {
+		if result := <-euchan; result.Err == nil {
 			c.Err = model.NewAppError("signupCompleteOAuth", "Team "+team.DisplayName+" already has a user with the email address attached to your "+service+" account", "email="+user.Email)
 			return
 		}
 
 		user.TeamId = team.Id
 
-		page := NewHtmlTemplatePage("signup_user_oauth", "Complete User Sign Up")
```

Comment:
Nope it has been removed in this PR since we want to just force pick a username for the user and the user can change it once they're in the system if they want to.